pkg/manifests/config.go

```diff
@@ -197,6 +199,24 @@ func NewConfig(content io.Reader) (*Config, error) {
 	res := &c
 	res.applyDefaults()
 	c.UserWorkloadConfiguration = NewDefaultUserWorkloadMonitoringConfig()
+	// The operator should only create some manifests if techPreview is enabled
+	c.TechPreview = tp
+
+	if c.ClusterMonitoringConfiguration.PrometheusK8sConfig.ScrapeProfile != FullScrapeProfile && !tp {
+		return nil, errors.Wrap(ErrConfigValidation, "scrapeProfiles is a TechPreview feature, to be able to use a profile different from the default (\"full\") please enable TechPreview")
+	}
+	// Validate ScrapeProfile field
+	foundProfile := false
+	for _, profile := range ScrapeProfiles {
+		if profile == c.ClusterMonitoringConfiguration.PrometheusK8sConfig.ScrapeProfile {
+			foundProfile = true
+			continue
+		}
+	}
+	if !foundProfile {
+		return nil, errors.Wrap(ErrConfigValidation, fmt.Sprintf(`scrape profile provided is unknown, supported scrape profiles are: %v`, ScrapeProfiles))
```

Comment:
nit: print the config-provided value
```suggestion
		return nil, errors.Wrap(ErrConfigValidation, fmt.Sprintf(`scrape profile %q is not supported, supported scrape profiles are: %q`, c.ClusterMonitoringConfiguration.PrometheusK8sConfig.ScrapeProfile, strings.Join(ScrapeProfiles, ", ")))
```